Automattic/pocket-casts-android
# Gravatar native Quick Editor

## Description
With the new [Gravatar SDK release](https://github.com/Automattic/Gravatar-SDK-Android/releases/tag/2.1.0), we can use the native Quick Editor (QE), which allows an easy avatar change without needing the browser (except for the OAuth authorization).

Currently, we are using the web version; this PR is replacing it with the native version included within the Gravatar SDK.

**Note:** The "official" `gravatarAppId` is already in the secrets, see [3fa232a](https://github.com/Automattic/pocket-casts-android/pull/3055/commits/3fa232a903f229783fab694c00146b14ccd46467).

## Testing Instructions

To test the OAuth Flow, you need a `gravatarAppId` in your `secret.properties`. You can create and configure your test app following [this](https://github.com/Automattic/Gravatar-SDK-Android/blob/trunk/docs/get-started/get-started.md#quick-editor). Let me know if you need help.

1. Enable `GRAVATAR_NATIVE_QUICK_EDITOR` feature flag under Profile -> Settings -> Beta features
2. Tap on the Profile tab
3. Tap on `Change avatar`
4. Verify Gravatar OAuth flow is opened in the browser
5. Complete the OAuth flow
6. You should be back on the app, and the QE should show your avatars
7. Modify your avatar
8. Close the QE and verify the new avatar is already shown in your account
9. Tap on `Change avatar`
10. OAuth flow should be presented again
11. Modify/upload your avatar
12. Close the QE and verify the new avatar is already shown in your account
13. `Sign out` from PocketCast
14. Login again with the same account
15. Tap on the Profile tab
16. Tap on `Change avatar`
17. OAuth flow should be presented again
18. Return to PocketCast
19. Disable `GRAVATAR_NATIVE_QUICK_EDITOR` feature flag under Profile -> Settings -> Beta features
20. Tap on the Profile tab
21. Tap on `Change avatar`
22. Browser with the web QE should be opened (previous behavior)

## Screenshots or Screencast 

<img width="300" alt="image" src="https://github.com/user-attachments/assets/ef722b66-e48f-44a8-8835-2e905071df4b"><img width="300" alt="image" src="https://github.com/user-attachments/assets/cb6cf3d2-a826-4f1a-b690-c2e0bda068da">

## Checklist
 
#### I have tested any UI changes...
